### PR TITLE
extend gamestar extractor to cover gamepro.de

### PR DESCRIPTION
**Pull request by lorem-ipsum**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information
[gamepro.de](gamepro.de) is the console focused variant of gamestar.de. Both use the same page layout and the existing gamestar extractor works fine for gamepro. Issue #3384 requested support for it a while ago.

### CODE REVIEW

File: youtube_dl/extractor/gamestar.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 
 class GameStarIE(InfoExtractor):
-    _VALID_URL = r'https?://(?:www\.)?gamestar\.de/videos/.*,(?P<id>[0-9]+)\.html'
+    _VALID_URL = r'https?://(?:www\.)?game(pro|star)\.de/videos/.*,(?P<id>[0-9]+)\.html'
```
Comment (dstftw) [inline youtube_dl/extractor/gamestar.py]: Don't capture groups you don't use. Add a test.

Comment (lorem-ipsum) [reply youtube_dl/extractor/gamestar.py]: Did both. Also fixed view_count and comment_count.

File: youtube_dl/extractor/gamestar.py [hunk 2]
```diff
@@ -38,11 +54,12 @@ def _real_extract(self, url):
             webpage, 'JSON-LD', group='json_ld'), video_id)
         info_dict = self._json_ld(json_ld, video_id)
         info_dict['title'] = remove_end(info_dict['title'], ' - GameStar')
+        info_dict['title'] = remove_end(info_dict['title'], ' - GamePro')
 
-        view_count = json_ld.get('interactionCount')
+        view_count = int(json_ld.get('interactionCount'))
```
Comment (dstftw) [inline youtube_dl/extractor/gamestar.py]: int_or_none.

File: youtube_dl/extractor/gamestar.py [hunk 3]
```diff
@@ -9,21 +9,37 @@
...
+            'info_dict': {
+                'id': '95316',
+                'ext': 'mp4',
+                'title': 'Top 10: Indie-Spiele für Nintendo Switch - Video: Tolle Nindies-Games zum Download',
+                'description': 'Wer behauptet, im letzten Jahr seien nicht ausreichend Spiele für die Switch\xa0herausgekommen, der hat offensichtlich nicht richtig...',
+                'thumbnail': r're:^https?://.*\.jpg$',
+                'timestamp': 1515173400,
+                'upload_date': '20180105',
+                'duration': 615,
+            }
```
Comment (dstftw) [inline youtube_dl/extractor/gamestar.py]: Make only_matching.

Comment (lorem-ipsum) [reply youtube_dl/extractor/gamestar.py]: I've removed everything except the URL and added the only_matching tag. Is that as desired?

File: youtube_dl/extractor/gamestar.py (hunk 2 above)
Comment (lorem-ipsum) [reply youtube_dl/extractor/gamestar.py]: done

File: youtube_dl/extractor/gamestar.py [hunk 4]
```diff
@@ -38,11 +44,12 @@ def _real_extract(self, url):
             webpage, 'JSON-LD', group='json_ld'), video_id)
         info_dict = self._json_ld(json_ld, video_id)
         info_dict['title'] = remove_end(info_dict['title'], ' - GameStar')
+        info_dict['title'] = remove_end(info_dict['title'], ' - GamePro')
```
Comment (dstftw) [inline youtube_dl/extractor/gamestar.py]: 46-47 code duplication.

Comment (lorem-ipsum) [reply youtube_dl/extractor/gamestar.py]: Any preferences on how to solve this? I can squeeze it in one line with two calls to `remove_end`, but in my eyes this make things only worse.

Comment (dstftw) [reply youtube_dl/extractor/gamestar.py]: The point is eliminating code duplication, not squeezing into one line.

Comment (lorem-ipsum) [reply youtube_dl/extractor/gamestar.py]: I don't really get what you want. I could remove the code duplication by defining a micro-function calling remove_end, or I could write my own remove_end which removing based on a regex or something like this. In my eyes neither change provides any benefit.
Please give me a direction on how I should reformat the code, I think we have different opinions on coding style.

**Pull request closed without merging**